activerecord/test/cases/adapters/postgresql/postgresql_adapter_test.rb

```diff
@@ -485,13 +485,16 @@ def test_unparsed_defaults_are_at_least_set_when_saving
       end
 
       def test_only_check_for_insensitive_comparison_capability_once
+        table_name = "ex"
         @connection.execute("CREATE DOMAIN example_type AS integer")
+        @connection.schema_cache.clear_data_source_cache!(table_name)
 
         with_example_table "id SERIAL PRIMARY KEY, number example_type" do
           number_klass = Class.new(ActiveRecord::Base) do
-            self.table_name = "ex"
+            self.table_name = table_name
           end
           attribute = number_klass.arel_table[:number]
+          @case_insensitive_cache = nil
```

Comment:
This particular line won't do what we need, because it's setting an instance variable in the wrong object.

> 2) @case_insensitive_cache not being cleared between tests

Is this currently a problem? It's a horrible hack, but my intention in https://github.com/rails/rails/compare/rails:3349c3b...rails:fcf3388 was to side-step this issue by giving the column a unique-to-this-test type name.

Reply:
Hey @matthewd 👋 , this isn't currently a problem but we see now that we were maybe future-proofing for something that has a low probability (but still a chance) of happening in that another test could be added in the future also using `example_type` as the column_type. 

We've removed our proposed changes to clearing `@case_insensitive_cache` in the tests so that our PR focuses solely on clearing the `schema_cache`.